toolkit/tools/pkg/imagecustomizerlib/kernelmoduleutils_test.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnsureModulesLoaded(t *testing.T) {
+	if os.Geteuid() != 0 {
+		t.Skip("Test must be run as root because it uses a chroot")
+	}
+
+	// Setup environment.
+	proposedDir := filepath.Join(tmpDir, "TestEnsureModulesLoaded")
+	chroot := safechroot.NewChroot(proposedDir, false)
+	err := chroot.Initialize("", []string{}, []*safechroot.MountPoint{}, false)
+	assert.NoError(t, err)
+	defer chroot.Close(false)
+
+	modulesLoadPath := filepath.Join(chroot.RootDir(), "etc/modules-load.d")
+	moduleLoadFilePath := filepath.Join(modulesLoadPath, "modules.conf")
+
+	// Only create parent directory. Test case where the file does not exist
+	err = os.MkdirAll(modulesLoadPath, os.ModePerm)
+	assert.NoError(t, err)
+	err = ensureModulesLoaded([]string{"module1"}, moduleLoadFilePath)
+	assert.NoError(t, err)
+
+	// Verify the file was created and contains the module name
+	content, err := os.ReadFile(moduleLoadFilePath)
+	assert.NoError(t, err)
+	assert.Contains(t, string(content), "module1\n")
+
+	// Test case where the file exists and already contains the module
+	err = ensureModulesLoaded([]string{"module1"}, moduleLoadFilePath)
+	assert.NoError(t, err)
+
+	// Verify the file content is unchanged (module not duplicated)
+	content, err = os.ReadFile(moduleLoadFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "module1\n", string(content))
+
+	// Test case where a new module is added
+	err = ensureModulesLoaded([]string{"module2"}, moduleLoadFilePath)
+	assert.NoError(t, err)
+
+	// Verify the file content now includes both modules
+	content, err = os.ReadFile(moduleLoadFilePath)
+	assert.NoError(t, err)
+	assert.Equal(t, "module1\nmodule2\n", string(content))
+}
+
+func TestEnsureModulesDisabled(t *testing.T) {
+	if os.Geteuid() != 0 {
+		t.Skip("Test must be run as root because it uses a chroot")
+	}
+
+	// Setup environment.
+	proposedDir := filepath.Join(tmpDir, "TestEnsureModulesDisabled")
+	chroot := safechroot.NewChroot(proposedDir, false)
+	err := chroot.Initialize("", []string{}, []*safechroot.MountPoint{}, false)
+	assert.NoError(t, err)
+	defer chroot.Close(false)
+
+	modprobePath := filepath.Join(chroot.RootDir(), "etc/modprobe.d")
+	moduleDisableFilePath := filepath.Join(modprobePath, "blacklist.conf")
+	err = os.MkdirAll(modprobePath, os.ModePerm)
+	assert.NoError(t, err)
+
+	// Prepopulate the file with a blacklisted module
```

Comment:
nit: `disabled module`

We should avoid using the word "blacklist" in our own code/comments.